header.go

```diff
@@ -54,11 +57,18 @@ func invoiceNumber(inv *bill.Invoice) string {
 // - 384 (Corrected invoice)
 // - 389 (Self-billed invoice)
 // - 381 (Credit note)
-func invoiceTypeCode(t cbc.Key) string {
+func invoiceTypeCode(inv *bill.Invoice) string {
+	if isSelfBilledInvoice(inv) {
+		return "389"
+	}
 	hash := map[cbc.Key]string{
 		bill.InvoiceTypeStandard:   "380",
 		bill.InvoiceTypeCorrective: "384",
 		bill.InvoiceTypeCreditNote: "381",
 	}
-	return hash[t]
+	return hash[inv.Type]
+}
+
+func isSelfBilledInvoice(inv *bill.Invoice) bool {
+	return inv.Tax != nil && inv.Type == bill.InvoiceTypeStandard && slices.Contains(inv.Tax.Tags, tax.TagSelfBilled)
```

Comment:
There are some extra helpers in GOBL already:
```suggestion
	return inv.Tax != nil && inv.Type == bill.InvoiceTypeStandard && tax.TagSelfBilled.In(inv.Tax.Tags...)
```
This looks great though!